scripts/release.js

```diff
@@ -236,6 +257,10 @@ if (process.platform == "linux") {
     }
     fs.writeFileSync(entitlementsPath, require("plist").build(entitlementsContents))
 
+    if (codesign) {
+        shell(`codesign -fs "${certName}" --deep ${appDirectory}`);
```

Comment:
Unfortunately the `--deep` strategy doesn't work for Onivim, because there's some nested dependencies it misses. We have a custom codesign script in: scripts/osx/codesign.sh

We'll probably  need to add a step to code-sign the sparkle framework there.

Reply:
Definitely! This is just for local debugging purposes, where it seems to work "okay". It definitely shouldn't be used for actual release codesigning, but I think it should be "good enough" for local debugging.

Reply:
I see - I think it's confusing that we have two code-signing strategies (it's already very complicated as-is 😬 ) - is it possible to use our existing code-sign strategy with a custom cert?

It seems like we could use the `OSX_P12_CERTIFICATE` and `CODESIGN_PASSWORD` environment variables with a self-signed certificate as well - this gets the benefit that we can verify that our production code-signing would work, too.

I'm also thinking about testing - I wonder if it would make sense to check-in a self-signed certificate, and then we could code-sign builds locally with that as part of release. If we had this - I could publish up some sort of dummy-update builds to a `test-self-signed` channel, that are signed with that same certificate - and we could have a test case to validate the update flow.

Reply:
It's tough to test the production flow right now - since it relies on both the current build and the updated build to be fully code-signed with our production certificate - but if we had a self-signed certificate available, we could test most of the flow with that!

Reply:
> I see - I think it's confusing that we have two code-signing strategies (it's already very complicated as-is 😬 ) - is it possible to use our existing code-sign strategy with a custom cert?

Sure!

> It's tough to test the production flow right now - since it relies on both the current build and the updated build to be fully code-signed with our production certificate - but if we had a self-signed certificate available, we could test most of the flow with that!

Sounds great -- I can check one in or you can, just let me know!

Reply:
Cool! Did you already generate one? If so - you could check-in that one, if you don't mind. We could add it in `scripts/osx` like `test-certificate` (just named clearly so people don't think we accidentally checked-in a production cert 😄 )

Reply:
Just checked in `test-certificate.p12` in `scripts/osx` with the password `OutrunLabs` (I also added a txt file with that information)!

Reply:
I just changed `release.js` to use `codesign.sh` with the new test certificate -- let me know if that's in line with what you were thinking!